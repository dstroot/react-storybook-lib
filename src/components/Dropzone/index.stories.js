import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

// components
import Dropzone from '../Dropzone';

// Dropzone section
storiesOf('Misc|Dropzone', module)
  .addParameters({
    // default for all stories in this book
    info: {
      inline: true,
      header: false,
      text: `
        Dropzone - drag and drop files to upload them.

        ##### References

        - https://malcoded.com/posts/react-dropzone
        - https://upmostly.com/tutorials/react-dropzone-file-uploads-react/
    `,
    },
  })
  .add('show Dropzone', () => <Dropzone onFilesAdded={action(console.log)} />);
