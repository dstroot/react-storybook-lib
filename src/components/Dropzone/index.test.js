import React from 'react';
import renderer from 'react-test-renderer';

import Dropzone from '../Dropzone';
// import { shallow, mount } from 'enzyme';
// import toJson from 'enzyme-to-json';

describe('Dropzone renders', () => {
  it('it should render', () => {
    const component = renderer.create(<Dropzone />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
