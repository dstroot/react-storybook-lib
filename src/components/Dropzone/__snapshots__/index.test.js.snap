// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropzone renders it should render 1`] = `
<div
  className="Dropzone "
  onClick={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <input
    className="FileInput"
    multiple={true}
    onChange={[Function]}
    type="file"
  />
  <img
    alt="upload"
    className="Icon"
    src="cloud.svg"
  />
  <span>
    Upload Files
  </span>
</div>
`;
